receiver/filelogreceiver/filelog_test.go

```diff
@@ -373,3 +374,77 @@ func (g *fileLogGenerator) Generate() []receivertest.UniqueIDAttrVal {
 	require.NoError(g.t, err)
 	return []receivertest.UniqueIDAttrVal{id}
 }
+
+func TestUTF16LEMultilineSAPAuditLog(t *testing.T) {
+	t.Parallel()
+
+	// Create a UTF-16LE encoded file with 10 SAP audit log records
+	// Each record starts with pattern: ([23])[A-Z][A-Z][A-Z0-9]\d{14}00
+	sapRecord := "2AUK20250227000000002316500018D110.102.8BATCH_ALRI                      SAPMSSY1                                0501Z91_VALR_IF&&Z91_VAL_PLSTATUS                                   10.122.81.29        "
+	
+	// Create 10 records concatenated (no newlines between them)
+	allRecords := ""
+	for i := 0; i < 10; i++ {
+		allRecords += sapRecord
+	}
+
+	// Encode to UTF-16LE
+	encoder := unicode.UTF16(unicode.LittleEndian, unicode.IgnoreBOM).NewEncoder()
+	encodedBuf := make([]byte, len(allRecords)*4) // UTF-16LE uses 2 bytes per ASCII char, allocate more for safety
+	nDst, _, err := encoder.Transform(encodedBuf, []byte(allRecords), true)
+	require.NoError(t, err)
+	encoded := encodedBuf[:nDst]
```

Comment:
Perhaps it would be simpler to use a writer with encoding from `golang.org/x/text/transform` so that we don't have to explicitly allocate memory? Something along these lines:

```suggestion
	file, err := os.Create("auditlog.txt")
	require.NoError(t, err)
	utf16Writer := transform.NewWriter(file, encoder)
	_, err = utf16Writer.Write([]byte(value))
```